src/http/url.cc

```diff
@@ -68,6 +81,35 @@ bool url_decode(const std::string_view& in, sstring& out) {
     return true;
 }
 
+sstring url_encode(const std::string_view& in) {
+    size_t encodable_chars = 0;
+    for (size_t i = 0; i < in.length(); i++) {
+        if (should_encode(in[i])) {
+            encodable_chars++;
+        }
+    }
+
+    if (encodable_chars == 0) {
+        return sstring(in);
+    }
+
+    sstring ret(in.length() + encodable_chars * 2 + 1, 0);
+    size_t o = 0;
+    for (size_t i = 0; i < in.length(); i++) {
+        if (in[i] == ' ') {
+            ret[o++] = '+';
```

Comment:
Why do you want to use `+` instead of `%20`? 
I guess both should work, but I wonder why you preferred `+` which is less natural (requires an extra special case).

Reply:
> Why do you want to use + instead of %20?

Less %XXs in the string -- easier to look at it :shrug:

Reply:
But strictly speaking, the RFC says that %20 should be used, not +, no?
I agree, thought, that + will work in all browser/HTML implementations.
But I think everything in the code will be simpler if you just used %20.